tangelo/algorithms/classical/fci_solver.py

```diff
@@ -131,3 +144,170 @@ def get_rdm(self):
+            warnings.warn("Degenerate orbitals are present in the molecule. The fci calculation is performed "
+                          "without symmetry so the rdms may not correspond to the integrals in molecule. A c1 "
+                          "version of the molecule with the correct integrals is present as FCISolverPsi4.molecule.")
+        one_rdm = np.asarray(self.ciwfn.get_opdm(0, 0, "SUM", False))
+        two_rdm = np.asarray(self.ciwfn.get_tpdm("SUM", True))
+
+        return one_rdm, two_rdm
+
+
+fci_solver_dict = {"pyscf": FCISolverPySCF, 'psi4': FCISolverPsi4}
+
+
+def get_fci_solver(molecule: SecondQuantizedMolecule, solver: Union[None, str, Type[ElectronicStructureSolver]] = default_fci_solver, **kwargs):
+    """Return requested target backend object.
+
+    Args:
+        solver (string or Type[ElectronicStructureSolver] or None): Supported string identifiers can be found in
+            fci_solver_dict (from tangelo.algorithms.classical.fci_solver). Can also provide a user-defined backend (child to ElectronicStructureSolver class)
+        kwargs: Other arguments that could be passed to a target. Examples are qubits_to_use for a QPU, transpiler
+            optimization level, error mitigation flags etc.
+     """
+
+    if solver is None:
+        if isinstance(molecule.solver, IntegralSolverPySCF):
+            solver = FCISolverPySCF
+        elif isinstance(molecule.solver, IntegralSolverPsi4):
+            solver = FCISolverPsi4
+        elif default_fci_solver is not None:
+            solver = default_fci_solver
+        else:
+            raise ModuleNotFoundError(f"One of the backends for {list(fci_solver_dict.keys())} needs to be installed to use FCISolver in Tangelo"
+                                      "without providing own implementation.")
+    # If target is a string use target_dict to return built-in backend
+    elif isinstance(solver, str):
+        try:
+            solver = fci_solver_dict[solver.lower()]
+        except KeyError:
+            raise ValueError(f"Error: backend {solver} not supported. Available built-in options: {list(fci_solver_dict.keys())}")
+    elif not issubclass(solver, ElectronicStructureSolver):
+        raise TypeError(f"Target must be a str or a subclass of Backend but received class {type(solver).__name__}")
+
+    return solver(molecule, **kwargs)
+
+
+# @deprecated("Please use get_fci_solver.")
+def FCISolver(molecule: SecondQuantizedMolecule, solver: Union[None, str, Type[ElectronicStructureSolver]] = default_fci_solver, **kwargs):
+    return get_fci_solver(molecule, solver, **kwargs)
+
+
+def getswaps(a):
+    """Takes a list and returns the swaps necessary to obtain the ordered version.
+
+    Example: getswaps([4, 3, 2, 1]) -> [[0, 3], [1, 2]]
+    swap [0, 3] results in [1, 3, 2, 4]
+    swap [1, 2] results in [1, 2, 3, 4]
+
+    Args:
+        a (list): Unordered list
+
+    Returns:
+        List[list]: The swap operations required to order the list.
```

Comment:
It would make more sense to return a list of 2-tuples. Each of this 2-tuple is a swap, an immutable unit of information. 

A list is usually for a mutable data-structure that can grow / shrink, where elements can be processed in any order to accomplish something. It is much more general and vague.